packages/lib/RotatingLogs.ts

```diff
@@ -0,0 +1,73 @@
+const moment = require('moment');
+const { FsDriverDummy } = require('./fs-driver-dummy.js');
+
+export default class RotatingLogs {
+
+	public static fsDriver_: any = null;
+
+	private logFileDir: string = '';
+	private logFileName: string = 'log.txt';
+	private logFileExpirationTimeInDays: number = 30;
+	private logFileMaximumSizeInBytes: number = 1024 * 1024 * 10;
+
+	public constructor(logFileDir: string) {
+		this.setLogFileDir(logFileDir);
+	}
+
+	public setLogFileDir(value: string) {
+		this.logFileDir = value;
+	}
+
+	private fsDriver() {
+		if (!RotatingLogs.fsDriver_) RotatingLogs.fsDriver_ = new FsDriverDummy();
+		return RotatingLogs.fsDriver_;
+	}
```

Comment:
You can use shim.fsDriver() directly in this class, so you don't need all this logic to set fsDriver_. You don't need an fsDriver_ property actually, just call shim.fsDriver()